app/actions/remote/channel.ts

```diff
@@ -438,38 +438,37 @@ export async function fetchMyChannel(serverUrl: string, teamId: string, channelI
 }
 
 export async function fetchMissingDirectChannelsInfo(serverUrl: string, directChannels: Channel[], locale?: string, teammateDisplayNameSetting?: string, currentUserId?: string, fetchOnly = false) {
-    const operator = DatabaseManager.serverDatabases[serverUrl]?.operator;
-    if (!operator) {
-        return {error: `${serverUrl} database not found`};
-    }
-
-    const {database} = operator;
-    const displayNameByChannel: Record<string, string> = {};
-    const users: UserProfile[] = [];
-    const updatedChannels = new Set<Channel>();
-
-    const dms: Channel[] = [];
-    const dmIds: string[] = [];
-    const dmWithoutDisplayName = new Set<string>();
-    const gms: Channel[] = [];
-    for (const c of directChannels) {
-        if (c.type === General.DM_CHANNEL) {
-            dms.push(c);
-            dmIds.push(c.id);
-            if (!c.display_name) {
-                dmWithoutDisplayName.add(c.id);
+    try {
+        const {database, operator} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+        const displayNameByChannel: Record<string, string> = {};
+        const users: UserProfile[] = [];
+        const updatedChannels = new Set<Channel>();
+
+        const dms: Channel[] = [];
+        const dmIds: string[] = [];
+        const dmWithoutDisplayName = new Set<string>();
+        const gms: Channel[] = [];
+        const channelIds = new Set(directChannels.map((c) => c.id));
+        const storedChannels = await queryChannelsById(database, Array.from(channelIds)).fetch();
+        const storedChannelsMap = new Map(storedChannels.map((c) => [c.id, c]));
+
+        for (const c of directChannels) {
+            if (c.type === General.DM_CHANNEL) {
+                dms.push(c);
+                dmIds.push(c.id);
+                if (!c.display_name && !storedChannelsMap.get(c.id)?.displayName) {
+                    dmWithoutDisplayName.add(c.id);
+                }
+                continue;
             }
-            continue;
+            gms.push(c);
```

Comment:
That is handled a bit differently in the `fetchProfilesInGroupChannels` function, do check it out and let me know

Reply:
The behaviour is better, since we can get many channel ids at the same time, but it is still requests to the server that may not be needed.

That being said, it helps keep things up to date, so I am ok leaving it like this for the time being.